rkt/metadata_service.go

```diff
@@ -75,6 +71,109 @@ func init() {
 	cmdMetadataService.Flags.BoolVar(&flagNoIdle, "no-idle", false, "exit when last container is unregistered")
 }
 
+type mdsPod struct {
+	uuid     types.UUID
+	ip       string
+	manifest *schema.PodManifest
+	apps     map[string]*schema.ImageManifest
+}
+
+type podStore struct {
+	byIP   map[string]*mdsPod
+	byUUID map[types.UUID]*mdsPod
+	mutex  sync.Mutex
```

Comment:
I know but podStore still has a set of methods that can be called on it, even if by this package. I'd like Lock/Unlock not to be among them. What's the benefit of embedding sync.Mutex in this case?